src/devices/vdml_imu.c

```diff
@@ -29,6 +29,17 @@ int32_t imu_reset(uint8_t port) {
 	return_port(port - 1, 1);
 }
 
+int32_t imu_set_data_rate(uint8_t port, uint32_t rate) {
+	claim_port_i(port - 1, E_DEVICE_IMU);
+	ERROR_IMU_STILL_CALIBRATING(port, device, PROS_ERR);
+
+	// rate is not less than 5ms, and rounded down to nearest increment of 5
+	rate -= rate == 0 ? 5 : (rate - 5) % 5;
```

Comment:
If rate is eg 3, then this won't work.

Reply:
C modulo takes the sign of the dividend, so you'd have 3 - ((3 - 5) % 5) -> 3 - (-2 % 5) -> 3 - (-2) -> 5

Reply:
I see now, yea. 🧠

Reply:
this is what i get for being clever i suppose